Description: Handle "=====" as another form of zero.
 gcov prints "=====" instead of "######" when an unexecuted line is
 "reachable only by exceptional paths such as C++ exception handlers."
 This should be handled the same as "######" for our purposes.
Author: Zack Weinberg <zackw@panix.com>
Last-Update: 2013-02-01

Index: lcov-1.12/bin/geninfo
===================================================================
--- a/bin/geninfo
+++ b/bin/geninfo
@@ -1771,8 +1771,9 @@ sub read_gcov_file($)
 				$number = (split(" ",substr($_, 0, 16)))[0];
 
 				# Check for zero count which is indicated
-				# by ######
-				if ($number eq "######") { $number = 0;	}
+				# by ###### or =====
+				if ($number eq "######" or
+				    $number eq "=====") { $number = 0; }
 
 				if ($exclude_line) {
 					# Register uninstrumented line instead
